test/BUILD

```diff
@@ -328,3 +328,11 @@ scala_specs2_junit_test(
     data = ["//src/java/com/google/devtools/build/lib:worker"],
     jvm_flags = ["-Dlocation.expanded=$(location //src/java/com/google/devtools/build/lib:worker)"],
 )
+
+scala_junit_test(
+    name = "JunitFiltersAbstractClassesAndInterfaces",
+    srcs = ["src/main/scala/scala/test/junit/JunitAbstractClassAndInterface.scala"],
+    size = "small",
+    suffixes = ["Test"],
+    print_discovered_classes = True
+)
```

Comment:
@ittaiz just to understand better - without your addition to the `DiscoveredTestSuite.scala` that test would cause exception, right? 
It might not be clear that this is what we're testing here. How can we explain that the runner **must filter out abstract and interfaces**? Maybe a comment?